api/api/settings.py

```diff
@@ -18,7 +18,7 @@
 # See https://docs.djangoproject.com/en/1.7/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = os.environ['DESECSTACK_API_SECRETKEY']
+SECRET_KEY = os.environ.get('DESECSTACK_API_SECRETKEY')
```

Comment:
> I see that the celery container is running the same image as the api. Can that result in trouble, for instance when one runs an image that is not up-to-date? Will docker-compose take care of automatically replacing the container if the image gets updated?

docker-compose should replace the container if the image gets updated, just like in the case of the `api` container (there is nothing special about the `api` container).

The celery container needs to run the same codebase so that the queued task can actually run "within our application". Celery just schedules application procedure calls, more or less.

> Also, will we be able to run other tasks through celery, too? Like, let's say, we want to defer an expensive/long running user action and send an email once it is done. Also, can we send email from tasks that run in celery?

Generally, that should be possible. We may have to configure different queues for different purposes with different properties (e.g. number of tasks picked up per minute etc.).

> Still, this lgtm and can be merged at your discretion.

:+1: Before I merge this, I would like your feedback on the one inline conversation that is left open at the moment. I'll add a comment there right now.

Reply:
The problem is that if we require these settings to be present, then we also have to provide them to the celery container although it does not need them for the tasks we have planned at the moment (sending out emails with fixed parameters).

It appears to me that especially secrets should be exposed only where necessary to minimize attack surface. This applies to the Django secret as well as the pdns API keys in particular.

On the other hand, I see your point. Maybe we could somehow determine which container we're running in, and raise an error given certain circumstances. However, that does not seem very clean to me. What are your thoughts?

Reply:
While minimizing exposure of sensitive information is important, that strategy here will run in trouble as soon as these values are needed by tasks run through celery. Let's say an email should contain a signed value, then the celery container executing the signature will need to know the secret key.

Reply:
as per discussion, reverted this change and added corresponding environment variables to the celery container (in the respective commit)